File: src/main/java/cpsc224/weapons/Weapon.java
package cpsc224.weapons;

import cpsc224.moves.Move;

public class Weapon {
    protected String name;
    protected int tier;
    


    private String name;
    private int tier;
    private Move move;

    public Weapon(int tier, String name, Move move) {
        this.tier = tier;
        this.name = name;
        this.move = move;
    }

    public String getName() {
        return name;
    }

    public int getTier() {
        return tier;
    }

    public Move getMove() {
        return move;
    }
}

File: web/src/game/items/weapons/Weapon.ts
import { Item } from '../Item';
import type { IconRef } from '../../utils/icons';
import type { Move } from './moves/Move';

export class Weapon extends Item {
  private move: Move;

  constructor(name: string, tier: number, move: Move, icon: IconRef = null) {
    super(name, tier, icon);
    this.move = move;
    if (move) move.setTier(tier);
  }

  getMove(): Move {
    return this.move;
  }

  override toString(): string {
    if (this.move.getMaxUses() < 0) return this.name;
    return `${this.name} (${this.move.getUses()})`;
  }

  getToolTipText(): string {
    const m = this.move;
    const uses = m.getMaxUses() < 0 ? '∞' : `${m.getUses()}/${m.getMaxUses()}`;
    return `${this.name} (T${this.tier})\n${m.getName()}: ${m.getDamage()} ${m.getDamageType()} dmg\nAcc: ${m.getAccuracy()} Uses: ${uses}`;
  }
}
